src/components/tl/mlx5/tl_mlx5.c

```diff
@@ -92,6 +92,10 @@ static ucc_config_field_t ucc_tl_mlx5_lib_config_table[] = {
      ucc_offsetof(ucc_tl_mlx5_lib_config_t, mcast_conf.max_eager),
      UCC_CONFIG_TYPE_MEMUNITS},
 
+    {"MCAST_DEVICE_MEM_ENABLE", "0", "Enable GPU memory support for Mcast",
```

Comment:
Why would a user want to disable the support?

Also, correct me if I'm wrong, but, from the code, I understand that this flag doesn't enable the "support", but rather forces the "usage" of GPU memory

Reply:
For a team, we can either have support for GPU buffers or CPU. We do not support both memory types at the same time. So if this env var is disabled, then it means we are supporting CPU buffers.

Reply:
Restriction comes from the resources that we have. We have a single UD QP per process, which can be used either for CUDA memory or HOST memory. In the future, we can come up with new designs that use double QP, so that we can have support for both. 
To check this, please see the new commit. We have a check to see if the mcast is cuda-aware, therefore, only running bcast-mcast on cuda memory.

Reply:
I am not sure to understand.

> We have a single UD QP per process, which can be used either for CUDA memory or HOST memory.

Yes, that's ok. Btw the creation of that QP is not in this PR. I am not suggesting to add an extra QP.

My point is Why `enable_cuda` is an option at the library level, while the user normally sets the memory type in UCC at the collective level.

I am saying this because I am afraid the current way is breaking UCC's requirements.

Reply:
can you please clarify which UCC requirement you are referring to?

_My point is Why enable_cuda is an option at the library level, while the user normally sets the memory type in UCC at the collective level._
if enable_cuda is set, then it means that the newly created mcast team only supports cuda buffers. If it is set to 0, then it means that the new mcast team only supports host buffers. Therefore, when a new collective is called, we need to check its mem type and compare it with the capabilities of the mcast team it is using.